library/src/main/java/com/datatorrent/lib/db/jdbc/AbstractJdbcPollInputOperator.java

```diff
@@ -269,20 +253,38 @@ public void deactivate()
     store.disconnect();
   }
 
+  protected void insertResultsInQueue(PreparedStatement preparedStatement) throws SQLException, InterruptedException
+  {
+    preparedStatement.setFetchSize(getFetchSize());
+    ResultSet result = preparedStatement.executeQuery();
+    if (result.next()) {
+      do {
+        while (!emitQueue.offer(getTuple(result))) {
+          Thread.sleep(DEFAULT_SLEEP_TIME);
+        }
+      } while (result.next());
+      result.close();
+    }
+    preparedStatement.close();
+  }
+
   protected void pollRecords()
   {
     if (isPolled) {
       return;
     }
     try {
-      ps.setFetchSize(getFetchSize());
-      ResultSet result = ps.executeQuery();
-      if (result.next()) {
-        do {
-          while (!emitQueue.offer(getTuple(result))) {
-            Thread.sleep(DEFAULT_SLEEP_TIME);
-          }
-        } while (result.next());
+      if (isPollerPartition) {
+        nextOffset = getRecordsCount();
+        while (lastOffset < nextOffset) {
+          PreparedStatement preparedStatement = store.getConnection().prepareStatement(buildRangeQuery(lastOffset, rangeSize),
```

Comment:
Basically the loop should continue until all the records currently in the table have been emitted.